feat(spanner): add support for multiregion CMEK

harshachinta opened the pull request:
Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
- [ ] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/java-spanner/issues/new/choose) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [ ] Ensure the tests and linter pass
- [ ] Code coverage does not decrease (if any source code was changed)
- [ ] Appropriate docs were updated (if necessary)

Fixes #<issue_number_goes_here> ☕️

If you write sample code, please follow the [samples format](
https://github.com/GoogleCloudPlatform/java-docs-samples/blob/main/SAMPLE_FORMAT.md).

harshachinta committed:
feat(spanner): add proto changes for multi region cmek

harshachinta committed:
feat(spanner): add code changes for multi region cmek

rajatbhatta commented on google-cloud-spanner/src/main/java/com/google/cloud/spanner/encryption/CustomerManagedEncryption.java:
```diff
@@ -17,29 +17,46 @@
   CustomerManagedEncryption(String kmsKeyName) {
     this.kmsKeyName = kmsKeyName;
+    // TODO(harsha): Should this be [] by default?
+    this.kmsKeyNames = com.google.protobuf.LazyStringArrayList.EMPTY;
```
```suggestion
    this.kmsKeyNames = new ArrayList<>();
```

rajatbhatta commented on google-cloud-spanner/src/main/java/com/google/cloud/spanner/encryption/CustomerManagedEncryption.java:
```diff
@@ -17,29 +17,46 @@
+    // TODO(harsha): Should this be "" by default?
+    this.kmsKeyName = "";
   }
 
```
Maybe we can have CustomerManagedEncryption(String... kmsKeyNames), and check if only one keyName is being passed in, or multiple?

harshachinta committed:
fix: update samples/snippets pom.xml configuration to avoid fat jar (#2100)

* fix: update samples/snippets pom.xml configuration to avoid fat jar

* fix: refactor path of jar creation and remove assembly-descriptor.xml file

* 🦉 Updates from OwlBot post-processor

See https://github.com/googleapis/repo-automation-bots/blob/main/packages/owl-bot/README.md

Co-authored-by: Owl Bot <gcf-owl-bot[bot]@users.noreply.github.com>

olavloite committed:
test: mock Spanner did not create jsonb parameters (#2182)

The mock Spanner server that is used for testing did not create statements
with PG JSONB typed parameters. Instead, all parameters with TypeCode JSON
would create statements with parameters of type JSON, regardless of any
TypeAnnotationCode.

gcf-owl-bot[bot] committed:
chore: upgrade native image checks to graalvm-22.3.0  (#1718) (#2177)

chore: upgrade native image checks to graalvm-22.3.0
Source-Link: https://github.com/googleapis/synthtool/commit/5e52896b69b145a3fac6827e4e75e617abdc9dd8
Post-Processor: gcr.io/cloud-devrel-public-resources/owlbot-java:latest@sha256:27b1b1884dce60460d7521b23c2a73376cba90c0ef3d9f0d32e4bdb786959cfd

Co-authored-by: Owl Bot <gcf-owl-bot[bot]@users.noreply.github.com>

rajatbhatta committed:
feat: setting up 6.33.x branch (#2184)

enable releases

ko3a4ok committed:
fix: use a proper endpoint for DirectPath tests (#2186)

* fix: use a proper endpoint for DirectPath tests

gauravpurohit06 committed:
refactor: Removing duplicated code to handle array values (#2143)

* refactor: Removing duplicated code to handle array values

The same deserialization logic for data types have been duplicated at multiple places which is prone to error.

* 🦉 Updates from OwlBot post-processor

See https://github.com/googleapis/repo-automation-bots/blob/main/packages/owl-bot/README.md

* refactor: Using lazy computation for all the data types and adding docs

Co-authored-by: Owl Bot <gcf-owl-bot[bot]@users.noreply.github.com>

gauravpurohit06 committed:
chore: Adding Kokoro configuration to run integration test against staging  (#2189)

chore: Adding Kokoro configuration to run integration test against staging

release-please[bot] committed:
chore(main): release 6.34.0 (#2169)

:robot: I have created a release *beep* *boop*
---


## [6.34.0](https://togithub.com/googleapis/java-spanner/compare/v6.33.0...v6.34.0) (2022-12-12)


### Features

* Setting up 6.33.x branch ([#2184](https://togithub.com/googleapis/java-spanner/issues/2184)) ([e237a21](https://togithub.com/googleapis/java-spanner/commit/e237a213cf5cb5edc338ca4e5f8ad5dd0593d2d1))


### Bug Fixes

* Remove the statement of session number limits ([#1928](https://togithub.com/googleapis/java-spanner/issues/1928)) ([ddd0625](https://togithub.com/googleapis/java-spanner/commit/ddd062527674659ca2ea73e079bca4dee62ca67f)), closes [#1927](https://togithub.com/googleapis/java-spanner/issues/1927)
* Update samples/snippets pom.xml configuration to avoid fat jar ([#2100](https://togithub.com/googleapis/java-spanner/issues/2100)) ([19058b4](https://togithub.com/googleapis/java-spanner/commit/19058b4cd324ce33e8dd52447bde2486c87d4754))
* Use a proper endpoint for DirectPath tests ([#2186](https://togithub.com/googleapis/java-spanner/issues/2186)) ([4d74a0d](https://togithub.com/googleapis/java-spanner/commit/4d74a0d8ae48e190c126ab4047b81cca117f4de1))


### Dependencies

* Update dependency org.graalvm.buildtools:junit-platform-native to v0.9.18 ([#2171](https://togithub.com/googleapis/java-spanner/issues/2171)) ([f348780](https://togithub.com/googleapis/java-spanner/commit/f3487805fe5f976596e94047c3796bc623eeae95))
* Update dependency org.graalvm.buildtools:native-maven-plugin to v0.9.18 ([#2145](https://togithub.com/googleapis/java-spanner/issues/2145)) ([dcdd2c3](https://togithub.com/googleapis/java-spanner/commit/dcdd2c3b684e38892fac0abbdf06081e9c7d83b2))

---
This PR was generated with [Release Please](https://togithub.com/googleapis/release-please). See [documentation](https://togithub.com/googleapis/release-please#release-please).

release-please[bot] committed:
chore(main): release 6.34.1-SNAPSHOT (#2196)

:robot: I have created a release *beep* *boop*
---


### Updating meta-information for bleeding-edge SNAPSHOT release.

---
This PR was generated with [Release Please](https://togithub.com/googleapis/release-please). See [documentation](https://togithub.com/googleapis/release-please#release-please).

renovate-bot committed:
build(deps): update dependency com.google.cloud:google-cloud-shared-config to v1.5.5 (#2179)

rajatbhatta committed:
chore: make Partition class getters public (#2195)

Make getters in Partition class public so that they can be accessed outside com.google.cloud.spanner package.

renovate-bot committed:
deps: update dependency com.google.cloud:google-cloud-shared-dependencies to v3.1.0 (#2187)

Co-authored-by: Tomo Suzuki <suztomo@google.com>

mpeddada1 committed:
chore: add native image checks as required (#1930)

release-please[bot] committed:
chore(main): release 6.34.1 (#2199)

:robot: I have created a release *beep* *boop*
---


## [6.34.1](https://togithub.com/googleapis/java-spanner/compare/v6.34.0...v6.34.1) (2022-12-13)


### Dependencies

* Update dependency com.google.cloud:google-cloud-shared-dependencies to v3.1.0 ([#2187](https://togithub.com/googleapis/java-spanner/issues/2187)) ([4d9df2b](https://togithub.com/googleapis/java-spanner/commit/4d9df2bac3a2dd6c910ba5fdd466ccd43a226c7f))

---
This PR was generated with [Release Please](https://togithub.com/googleapis/release-please). See [documentation](https://togithub.com/googleapis/release-please#release-please).

release-please[bot] committed:
chore(main): release 6.34.2-SNAPSHOT (#2200)

:robot: I have created a release *beep* *boop*
---


### Updating meta-information for bleeding-edge SNAPSHOT release.

---
This PR was generated with [Release Please](https://togithub.com/googleapis/release-please). See [documentation](https://togithub.com/googleapis/release-please#release-please).

gauravpurohit06 committed:
chore: Updating Kokoro configuration (#2198)

* chore: Adding missing slash

* 🦉 Updates from OwlBot post-processor

See https://github.com/googleapis/repo-automation-bots/blob/main/packages/owl-bot/README.md

* 🦉 Updates from OwlBot post-processor

See https://github.com/googleapis/repo-automation-bots/blob/main/packages/owl-bot/README.md

Co-authored-by: Owl Bot <gcf-owl-bot[bot]@users.noreply.github.com>

renovate-bot committed:
test(deps): update dependency org.mockito:mockito-core to v4.10.0 (#2201)

[![Mend Renovate](https://app.renovatebot.com/images/banner.svg)](https://renovatebot.com)

This PR contains the following updates:

| Package | Change | Age | Adoption | Passing | Confidence |
|---|---|---|---|---|---|
| [org.mockito:mockito-core](https://togithub.com/mockito/mockito) | `4.9.0` -> `4.10.0` | [![age](https://badges.renovateapi.com/packages/maven/org.mockito:mockito-core/4.10.0/age-slim)](https://docs.renovatebot.com/merge-confidence/) | [![adoption](https://badges.renovateapi.com/packages/maven/org.mockito:mockito-core/4.10.0/adoption-slim)](https://docs.renovatebot.com/merge-confidence/) | [![passing](https://badges.renovateapi.com/packages/maven/org.mockito:mockito-core/4.10.0/compatibility-slim/4.9.0)](https://docs.renovatebot.com/merge-confidence/) | [![confidence](https://badges.renovateapi.com/packages/maven/org.mockito:mockito-core/4.10.0/confidence-slim/4.9.0)](https://docs.renovatebot.com/merge-confidence/) |

---

### Release Notes

<details>
<summary>mockito/mockito</summary>

### [`v4.10.0`](https://togithub.com/mockito/mockito/releases/tag/v4.10.0)

[Compare Source](https://togithub.com/mockito/mockito/compare/v4.9.0...v4.10.0)

<sup><sup>*Changelog generated by [Shipkit Changelog Gradle Plugin](https://togithub.com/shipkit/shipkit-changelog)*</sup></sup>

##### 4.10.0

-   2022-12-14 - [13 commit(s)](https://togithub.com/mockito/mockito/compare/v4.9.0...v4.10.0) by Andrei Solntsev, Andriy Redko, Andy Coates, Christopher Lambert, Marcono1234, Vladimir Glinskikh, dependabot\[bot]
-   Add new artifact mockito-subclass (to use mock-maker-subclass MockMaker) [(#&#8203;2821)](https://togithub.com/mockito/mockito/pull/2821)
-   Bump gradle from 7.5.1 to 7.6 [(#&#8203;2817)](https://togithub.com/mockito/mockito/pull/2817)
-   Fix incorrect Javadoc inline tag for MockitoJUnitRunner [(#&#8203;2816)](https://togithub.com/mockito/mockito/pull/2816)
-   Bump shipkit-auto-version from 1.2.1 to 1.2.2 [(#&#8203;2811)](https://togithub.com/mockito/mockito/pull/2811)
-   Bump com.github.ben-manes.versions from 0.42.0 to 0.44.0 [(#&#8203;2810)](https://togithub.com/mockito/mockito/pull/2810)
-   Bump kotlinVersion from 1.7.21 to 1.7.22 [(#&#8203;2809)](https://togithub.com/mockito/mockito/pull/2809)
-   Bump junit from 1.1.3 to 1.1.4 [(#&#8203;2806)](https://togithub.com/mockito/mockito/pull/2806)
-   Simplify `MatcherApplicationStrategy` [(#&#8203;2803)](https://togithub.com/mockito/mockito/pull/2803)
-   Bump kotlinVersion from 1.7.10 to 1.7.21 [(#&#8203;2801)](https://togithub.com/mockito/mockito/pull/2801)
-   Bump espresso-core from 3.4.0 to 3.5.0 [(#&#8203;2800)](https://togithub.com/mockito/mockito/pull/2800)
-   Bump versions.bytebuddy from 1.12.16 to 1.12.19 [(#&#8203;2799)](https://togithub.com/mockito/mockito/pull/2799)
-   Upgrade errorprone from 2.14.0 to 2.16 [(#&#8203;2794)](https://togithub.com/mockito/mockito/pull/2794)
-   automatically detect class to mock [(#&#8203;2779)](https://togithub.com/mockito/mockito/pull/2779)

</details>

---

### Configuration

📅 **Schedule**: Branch creation - At any time (no schedule defined), Automerge - At any time (no schedule defined).

🚦 **Automerge**: Disabled by config. Please merge this manually once you are satisfied.

♻ **Rebasing**: Whenever PR becomes conflicted, or you tick the rebase/retry checkbox.

🔕 **Ignore**: Close this PR and you won't be reminded about this update again.

---

 - [ ] <!-- rebase-check -->If you want to rebase/retry this PR, check this box

---

This PR has been generated by [Mend Renovate](https://www.mend.io/free-developer-tools/renovate/). View repository job log [here](https://app.renovatebot.com/dashboard#github/googleapis/java-spanner).
<!--renovate-debug:eyJjcmVhdGVkSW5WZXIiOiIzNC4yNC4wIiwidXBkYXRlZEluVmVyIjoiMzQuMjQuMCJ9-->

rahul2393 committed:
feat(spanner): add samples for fine grained access control (#2172)

* feat(spanner): add samples for fine grained access control

* 🦉 Updates from OwlBot post-processor

See https://github.com/googleapis/repo-automation-bots/blob/main/packages/owl-bot/README.md

* fix sample checkstyles

* incorporate review comment

* incorporate requested changes

* incorporate requested changes

* fix checkstyle

* incorporate review comments

* 🦉 Updates from OwlBot post-processor

See https://github.com/googleapis/repo-automation-bots/blob/main/packages/owl-bot/README.md

* refactoring

* remove role check & exceptions

Co-authored-by: Owl Bot <gcf-owl-bot[bot]@users.noreply.github.com>

renovate-bot committed:
chore(deps): update dependency com.google.cloud:libraries-bom to v26.2.0 (#2203)

* chore(deps): update dependency com.google.cloud:libraries-bom to v26.2.0

* 🦉 Updates from OwlBot post-processor

See https://github.com/googleapis/repo-automation-bots/blob/main/packages/owl-bot/README.md

Co-authored-by: Owl Bot <gcf-owl-bot[bot]@users.noreply.github.com>

renovate-bot committed:
deps: update dependency org.graalvm.buildtools:native-maven-plugin to v0.9.19 (#2181)

renovate-bot committed:
deps: update dependency org.graalvm.buildtools:junit-platform-native to v0.9.19 (#2180)

renovate-bot committed:
build(deps): update dependency org.apache.maven.plugins:maven-dependency-plugin to v3.4.0 (#2178)

renovate-bot committed:
chore(deps): update dependency com.google.cloud:google-cloud-spanner to v6.34.0 (#2197)

[![Mend Renovate](https://app.renovatebot.com/images/banner.svg)](https://renovatebot.com)

This PR contains the following updates:

| Package | Change | Age | Adoption | Passing | Confidence |
|---|---|---|---|---|---|
| [com.google.cloud:google-cloud-spanner](https://togithub.com/googleapis/java-spanner) | `6.33.0` -> `6.34.1` | [![age](https://badges.renovateapi.com/packages/maven/com.google.cloud:google-cloud-spanner/6.34.1/age-slim)](https://docs.renovatebot.com/merge-confidence/) | [![adoption](https://badges.renovateapi.com/packages/maven/com.google.cloud:google-cloud-spanner/6.34.1/adoption-slim)](https://docs.renovatebot.com/merge-confidence/) | [![passing](https://badges.renovateapi.com/packages/maven/com.google.cloud:google-cloud-spanner/6.34.1/compatibility-slim/6.33.0)](https://docs.renovatebot.com/merge-confidence/) | [![confidence](https://badges.renovateapi.com/packages/maven/com.google.cloud:google-cloud-spanner/6.34.1/confidence-slim/6.33.0)](https://docs.renovatebot.com/merge-confidence/) |

---

### Configuration

📅 **Schedule**: Branch creation - At any time (no schedule defined), Automerge - At any time (no schedule defined).

🚦 **Automerge**: Disabled by config. Please merge this manually once you are satisfied.

♻ **Rebasing**: Renovate will not automatically rebase this PR, because other commits have been found.

🔕 **Ignore**: Close this PR and you won't be reminded about this update again.

---

 - [ ] <!-- rebase-check -->If you want to rebase/retry this PR, check this box

---

This PR has been generated by [Mend Renovate](https://www.mend.io/free-developer-tools/renovate/). View repository job log [here](https://app.renovatebot.com/dashboard#github/googleapis/java-spanner).
<!--renovate-debug:eyJjcmVhdGVkSW5WZXIiOiIzNC4yNC4wIiwidXBkYXRlZEluVmVyIjoiMzQuMjQuMCJ9-->

renovate-bot committed:
chore(deps): update dependency com.google.cloud:google-cloud-spanner to v6.34.1 (#2204)

* chore(deps): update dependency com.google.cloud:google-cloud-spanner to v6.34.1

* 🦉 Updates from OwlBot post-processor

See https://github.com/googleapis/repo-automation-bots/blob/main/packages/owl-bot/README.md

Co-authored-by: Owl Bot <gcf-owl-bot[bot]@users.noreply.github.com>

gcf-owl-bot[bot] committed:
build(deps): bump certifi from 2022.9.24 to 2022.12.7 in /synthtool/gcp/templates/java_library/.kokoro (#1732) (#2194)

build(deps): bump certifi

Bumps [certifi](https://togithub.com/certifi/python-certifi) from 2022.9.24 to 2022.12.7.
- [Release notes](https://togithub.com/certifi/python-certifi/releases)
- [Commits](https://togithub.com/certifi/python-certifi/compare/2022.09.24...2022.12.07)

---
updated-dependencies:
- dependency-name: certifi
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: Jeff Ching <chingor@google.com>
Source-Link: https://togithub.com/googleapis/synthtool/commit/ae0d43e5f17972981fe501ecf5a5d20055128bea
Post-Processor: gcr.io/cloud-devrel-public-resources/owlbot-java:latest@sha256:9de537d592b60e5eac73b374a28263969bae91ecdb29b445e894576fbf54851c

renovate-bot committed:
deps: update dependency com.google.cloud:google-cloud-monitoring to v3.8.0 (#2192)

[![Mend Renovate](https://app.renovatebot.com/images/banner.svg)](https://renovatebot.com)

This PR contains the following updates:

| Package | Change | Age | Adoption | Passing | Confidence |
|---|---|---|---|---|---|
| [com.google.cloud:google-cloud-monitoring](https://togithub.com/googleapis/google-cloud-java) | `3.7.0` -> `3.8.0` | [![age](https://badges.renovateapi.com/packages/maven/com.google.cloud:google-cloud-monitoring/3.8.0/age-slim)](https://docs.renovatebot.com/merge-confidence/) | [![adoption](https://badges.renovateapi.com/packages/maven/com.google.cloud:google-cloud-monitoring/3.8.0/adoption-slim)](https://docs.renovatebot.com/merge-confidence/) | [![passing](https://badges.renovateapi.com/packages/maven/com.google.cloud:google-cloud-monitoring/3.8.0/compatibility-slim/3.7.0)](https://docs.renovatebot.com/merge-confidence/) | [![confidence](https://badges.renovateapi.com/packages/maven/com.google.cloud:google-cloud-monitoring/3.8.0/confidence-slim/3.7.0)](https://docs.renovatebot.com/merge-confidence/) |

---

### Configuration

📅 **Schedule**: Branch creation - At any time (no schedule defined), Automerge - At any time (no schedule defined).

🚦 **Automerge**: Disabled by config. Please merge this manually once you are satisfied.

♻ **Rebasing**: Whenever PR becomes conflicted, or you tick the rebase/retry checkbox.

🔕 **Ignore**: Close this PR and you won't be reminded about this update again.

---

 - [ ] <!-- rebase-check -->If you want to rebase/retry this PR, check this box

---

This PR has been generated by [Mend Renovate](https://www.mend.io/free-developer-tools/renovate/). View repository job log [here](https://app.renovatebot.com/dashboard#github/googleapis/java-spanner).
<!--renovate-debug:eyJjcmVhdGVkSW5WZXIiOiIzNC4yNC4wIiwidXBkYXRlZEluVmVyIjoiMzQuMjQuMCJ9-->

renovate-bot committed:
deps: update dependency com.google.cloud:google-cloud-trace to v2.7.0 (#2193)

[![Mend Renovate](https://app.renovatebot.com/images/banner.svg)](https://renovatebot.com)

This PR contains the following updates:

| Package | Change | Age | Adoption | Passing | Confidence |
|---|---|---|---|---|---|
| [com.google.cloud:google-cloud-trace](https://togithub.com/googleapis/google-cloud-java) | `2.6.0` -> `2.7.0` | [![age](https://badges.renovateapi.com/packages/maven/com.google.cloud:google-cloud-trace/2.7.0/age-slim)](https://docs.renovatebot.com/merge-confidence/) | [![adoption](https://badges.renovateapi.com/packages/maven/com.google.cloud:google-cloud-trace/2.7.0/adoption-slim)](https://docs.renovatebot.com/merge-confidence/) | [![passing](https://badges.renovateapi.com/packages/maven/com.google.cloud:google-cloud-trace/2.7.0/compatibility-slim/2.6.0)](https://docs.renovatebot.com/merge-confidence/) | [![confidence](https://badges.renovateapi.com/packages/maven/com.google.cloud:google-cloud-trace/2.7.0/confidence-slim/2.6.0)](https://docs.renovatebot.com/merge-confidence/) |

---

### Configuration

📅 **Schedule**: Branch creation - At any time (no schedule defined), Automerge - At any time (no schedule defined).

🚦 **Automerge**: Disabled by config. Please merge this manually once you are satisfied.

♻ **Rebasing**: Whenever PR becomes conflicted, or you tick the rebase/retry checkbox.

🔕 **Ignore**: Close this PR and you won't be reminded about this update again.

---

 - [ ] <!-- rebase-check -->If you want to rebase/retry this PR, check this box

---

This PR has been generated by [Mend Renovate](https://www.mend.io/free-developer-tools/renovate/). View repository job log [here](https://app.renovatebot.com/dashboard#github/googleapis/java-spanner).
<!--renovate-debug:eyJjcmVhdGVkSW5WZXIiOiIzNC4yNC4wIiwidXBkYXRlZEluVmVyIjoiMzQuMjQuMCJ9-->

dependabot[bot] committed:
build(deps): bump certifi from 2022.9.24 to 2022.12.7 in /.kokoro (#2190)

Bumps [certifi](https://togithub.com/certifi/python-certifi) from 2022.9.24 to 2022.12.7.
<details>
<summary>Commits</summary>
<ul>
<li><a href="https://togithub.com/certifi/python-certifi/commit/9e9e840925d7b8e76c76fdac1fab7e6e88c1c3b8"><code>9e9e840</code></a> 2022.12.07</li>
<li>See full diff in <a href="https://togithub.com/certifi/python-certifi/compare/2022.09.24...2022.12.07">compare view</a></li>
</ul>
</details>
<br />


[![Dependabot compatibility score](https://dependabot-badges.githubapp.com/badges/compatibility_score?dependency-name=certifi&package-manager=pip&previous-version=2022.9.24&new-version=2022.12.7)](https://docs.github.com/en/github/managing-security-vulnerabilities/about-dependabot-security-updates#about-compatibility-scores)

Dependabot will resolve any conflicts with this PR as long as you don't alter it yourself. You can also trigger a rebase manually by commenting `@dependabot rebase`.

[//]: # (dependabot-automerge-start)
[//]: # (dependabot-automerge-end)

---

<details>
<summary>Dependabot commands and options</summary>
<br />

You can trigger Dependabot actions by commenting on this PR:
- `@dependabot rebase` will rebase this PR
- `@dependabot recreate` will recreate this PR, overwriting any edits that have been made to it
- `@dependabot merge` will merge this PR after your CI passes on it
- `@dependabot squash and merge` will squash and merge this PR after your CI passes on it
- `@dependabot cancel merge` will cancel a previously requested merge and block automerging
- `@dependabot reopen` will reopen this PR if it is closed
- `@dependabot close` will close this PR and stop Dependabot recreating it. You can achieve the same result by closing it manually
- `@dependabot ignore this major version` will close this PR and stop Dependabot creating any more for this major version (unless you reopen the PR or upgrade to it yourself)
- `@dependabot ignore this minor version` will close this PR and stop Dependabot creating any more for this minor version (unless you reopen the PR or upgrade to it yourself)
- `@dependabot ignore this dependency` will close this PR and stop Dependabot creating any more for this dependency (unless you reopen the PR or upgrade to it yourself)
- `@dependabot use these labels` will set the current labels as the default for future PRs for this repo and language
- `@dependabot use these reviewers` will set the current reviewers as the default for future PRs for this repo and language
- `@dependabot use these assignees` will set the current assignees as the default for future PRs for this repo and language
- `@dependabot use this milestone` will set the current milestone as the default for future PRs for this repo and language

You can disable automated security fix PRs for this repo from the [Security Alerts page](https://togithub.com/googleapis/java-spanner/network/alerts).

</details>

gcf-owl-bot[bot] committed:
chore: Enable requesting numeric enums in "transport=rest" responses for services supporting this (Java, Go, Python, PHP, TypeScript, C#, and Ruby), even if they do not yet turn on REST transport (#2140)

* chore: Set `rest_numeric_enums = False` for all gapic rules explicitly

PiperOrigin-RevId: 485623855

Source-Link: https://github.com/googleapis/googleapis/commit/807125e7953e32535710924fe4bf0362ac93754d

Source-Link: https://github.com/googleapis/googleapis-gen/commit/c4ddd6d651da69658a4762c5eafff5cd8f0e25c5
Copy-Tag: eyJwIjoiLmdpdGh1Yi8uT3dsQm90LnlhbWwiLCJoIjoiYzRkZGQ2ZDY1MWRhNjk2NThhNDc2MmM1ZWFmZmY1Y2Q4ZjBlMjVjNSJ9

* 🦉 Updates from OwlBot post-processor

See https://github.com/googleapis/repo-automation-bots/blob/main/packages/owl-bot/README.md

* chore: Enable requesting numeric enums in "transport=rest" responses for services supporting this (Java, Go, Python, PHP, TypeScript, C#, and Ruby), even if they do not yet turn on REST transport

chore: disallow "transport=rest" for services where numeric enums are not confirmed to be supported (except in PHP and Java)
PiperOrigin-RevId: 493113566

Source-Link: https://github.com/googleapis/googleapis/commit/758f0d1217d9c7fe398aa5efb1057ce4b6409e55

Source-Link: https://github.com/googleapis/googleapis-gen/commit/78bd8f05e1276363eb14eae70e91fe4bc20703ab
Copy-Tag: eyJwIjoiLmdpdGh1Yi8uT3dsQm90LnlhbWwiLCJoIjoiNzhiZDhmMDVlMTI3NjM2M2ViMTRlYWU3MGU5MWZlNGJjMjA3MDNhYiJ9

* 🦉 Updates from OwlBot post-processor

See https://github.com/googleapis/repo-automation-bots/blob/main/packages/owl-bot/README.md

Co-authored-by: Owl Bot <gcf-owl-bot[bot]@users.noreply.github.com>
Co-authored-by: Lawrence Qiu <lqiu96@gmail.com>

harshachinta committed:
feat(spanner): code updates for multi-region cmek

harshachinta committed:
feat(spanner): testcase updates

mpeddada1 committed:
chore: fix kokoro windows java 8 ci by setting JAVA11_HOME env variable (#2207)

* chore: fix kokoro windows java 8 ci by setting JAVA11_HOME env variable

mpeddada1 committed:
chore: ensure that java8 is used to run tests in kokoro windows job (#2210)

rajatbhatta committed:
test: add tests for JSONB array support (#2209)

* test: add tests for JSONB array support

renovate-bot committed:
test(deps): update dependency org.mockito:mockito-core to v4.11.0 (#2213)

rahul2393 committed:
fix: retry on RST_STREAM internal error (#2111)

* fix: retry on RST_STREAM internal error

* Update google-cloud-spanner/src/test/java/com/google/cloud/spanner/IsRetryableInternalErrorTest.java

Co-authored-by: Knut Olav Løite <koloite@gmail.com>

* 🦉 Updates from OwlBot post-processor

See https://github.com/googleapis/repo-automation-bots/blob/main/packages/owl-bot/README.md

Co-authored-by: Knut Olav Løite <koloite@gmail.com>
Co-authored-by: Owl Bot <gcf-owl-bot[bot]@users.noreply.github.com>

gyang-google committed:
feat: add support for new cloud client test framework in google-cloud-spanner-executor (#2217)

* Add Spanner executor proto and implementation for Read/Mutation and admin actions.

* Add Spanner executor proto and implementation for Read/Mutation and admin actions.

* Add Spanner executor proto and implementation for Read/Mutation and admin actions.

* Add Spanner executor proto and implementation for Read/Mutation and admin actions.

* Add Spanner executor proto and implementation for Read/Mutation and admin actions.

* Fix error code mapping.

* Remove read_timestamp field in read result as it will never use it.(for optimistic only)

* Mark all fields in spanner action outcome as optional.

* Refactored existing protos according to api dev lint.
Prepare new protos for existing executor features.

* Added DML/batch DML/query features.
Commented out Batch txns and change stream for future support.

* Added support to Write.

* Update batch txn related protos.

* Add support for batch txn feature.

* Add metadata for batch read.
Prepare protos for change stream.

* Implement the support for change stream.

* Update change stream proto fields.

* Fix create cloud backup bug.

* Fix numeric related issue.
Fix partition token.
Fix request index for batch read.

* Make all calls to streamObserver synchronized to prevent concurrency issue. This is achieved by wrapping streamObserver into executionFlowContext and make all the calls synchronized.

* Rebase onto java-spanner changes.

* Add Spanner executor proto and implementation for Read/Mutation and admin actions.

* Add Spanner executor proto and implementation for Read/Mutation and admin actions.

* Add Spanner executor proto and implementation for Read/Mutation and admin actions.

* Add Spanner executor proto and implementation for Read/Mutation and admin actions.

* Fix error code mapping.

* Remove read_timestamp field in read result as it will never use it.(for optimistic only)

* Mark all fields in spanner action outcome as optional.

* Refactored existing protos according to api dev lint.
Prepare new protos for existing executor features.

* Added DML/batch DML/query features.
Commented out Batch txns and change stream for future support.

* Added support to Write.

* Update batch txn related protos.

* Add support for batch txn feature.

* Add metadata for batch read.
Prepare protos for change stream.

* Implement the support for change stream.

* Update change stream proto fields.

* Fix create cloud backup bug.

* Fix numeric related issue.
Fix partition token.
Fix request index for batch read.

* Make all calls to streamObserver synchronized to prevent concurrency issue. This is achieved by wrapping streamObserver into executionFlowContext and make all the calls synchronized.

* Replaces protos with the final released version.

* Apply maven-protobuf-plugin to generate grpc stubs, remove grpc folder for executor proto.
Add key file option to client creation.

* Remove partitionTestUtil since partitionToken can be fetched now.

* Remove partitionTestUtil since partitionToken can be fetched now.

* Cleanup.

* Fix pageToken for all the list operations.
Reformat code.

* Rephrase some comments.
Add createUserInstanceConfig and deleteUserInstanceConfig.
Fix instanceConfig conversion issue.

* Fix key flag issue.
Fix replicas in createUserInstanceConfig.

* Fix some admin actions regarding instanceId args.

* Remove unused stuff in metadata.
Fix numeric key issue.

* Remove debug message.

* Remove debug message.

* Update build files to accommodate with proto dependency changes.

* Update pom file.

* Update pom file.

* fix: update comments

* fix: clean up dependencies.

* 🦉 Updates from OwlBot post-processor

See https://github.com/googleapis/repo-automation-bots/blob/main/packages/owl-bot/README.md

* fix: addressed comments.

* 🦉 Updates from OwlBot post-processor

See https://github.com/googleapis/repo-automation-bots/blob/main/packages/owl-bot/README.md

Co-authored-by: Owl Bot <gcf-owl-bot[bot]@users.noreply.github.com>

gauravpurohit06 committed:
chore: Updating Integration tests service account (#2224)

renovate-bot committed:
deps: update dependency com.google.cloud:google-cloud-shared-dependencies to v3.1.1 (#2222)

* deps: update dependency com.google.cloud:google-cloud-shared-dependencies to v3.1.1

* 🦉 Updates from OwlBot post-processor

See https://github.com/googleapis/repo-automation-bots/blob/main/packages/owl-bot/README.md

Co-authored-by: Owl Bot <gcf-owl-bot[bot]@users.noreply.github.com>

gauravpurohit06 committed:
chore: Updating instance name (#2229)

Updating instance name

release-please[bot] committed:
chore(main): release 6.35.0 (#2202)

Co-authored-by: release-please[bot] <55107282+release-please[bot]@users.noreply.github.com>

release-please[bot] committed:
chore(main): release 6.35.1-SNAPSHOT (#2232)

:robot: I have created a release *beep* *boop*
---


### Updating meta-information for bleeding-edge SNAPSHOT release.

---
This PR was generated with [Release Please](https://togithub.com/googleapis/release-please). See [documentation](https://togithub.com/googleapis/release-please#release-please).

gcf-owl-bot[bot] committed:
chore: regenerate API index (#2221)

* chore: regenerate API index

Source-Link: https://github.com/googleapis/googleapis/commit/b84b4fe3d9ad3ef8c06aaa00114f53061169c531

Source-Link: https://github.com/googleapis/googleapis-gen/commit/916424c97215b12211037b5057b833447d073bbd
Copy-Tag: eyJwIjoiLmdpdGh1Yi8uT3dsQm90LnlhbWwiLCJoIjoiOTE2NDI0Yzk3MjE1YjEyMjExMDM3YjUwNTdiODMzNDQ3ZDA3M2JiZCJ9

* 🦉 Updates from OwlBot post-processor

See https://github.com/googleapis/repo-automation-bots/blob/main/packages/owl-bot/README.md

Co-authored-by: Owl Bot <gcf-owl-bot[bot]@users.noreply.github.com>

renovate-bot committed:
deps: update dependency org.junit.vintage:junit-vintage-engine to v5.9.2 (#2223)

renovate-bot committed:
build(deps): update dependency org.apache.maven.plugins:maven-project-info-reports-plugin to v3.4.2 (#2225)

renovate-bot committed:
build(deps): update dependency org.apache.maven.plugins:maven-failsafe-plugin to v3.0.0-m8 (#2226)

renovate-bot committed:
deps: update dependency com.google.cloud:google-cloud-monitoring to v3.9.0 (#2230)

renovate-bot committed:
chore(deps): update dependency com.google.cloud:google-cloud-spanner to v6.35.0 (#2233)

* chore(deps): update dependency com.google.cloud:google-cloud-spanner to v6.35.0

* 🦉 Updates from OwlBot post-processor

See https://github.com/googleapis/repo-automation-bots/blob/main/packages/owl-bot/README.md

Co-authored-by: Owl Bot <gcf-owl-bot[bot]@users.noreply.github.com>

renovate-bot committed:
build(deps): update dependency org.apache.maven.plugins:maven-dependency-plugin to v3.5.0 (#2227)

renovate-bot committed:
chore(deps): update dependency com.google.cloud:libraries-bom to v26.3.0 (#2220)

[![Mend Renovate](https://app.renovatebot.com/images/banner.svg)](https://renovatebot.com)

This PR contains the following updates:

| Package | Change | Age | Adoption | Passing | Confidence |
|---|---|---|---|---|---|
| [com.google.cloud:libraries-bom](https://cloud.google.com/java/docs/bom) ([source](https://togithub.com/googleapis/java-cloud-bom)) | `26.2.0` -> `26.3.0` | [![age](https://badges.renovateapi.com/packages/maven/com.google.cloud:libraries-bom/26.3.0/age-slim)](https://docs.renovatebot.com/merge-confidence/) | [![adoption](https://badges.renovateapi.com/packages/maven/com.google.cloud:libraries-bom/26.3.0/adoption-slim)](https://docs.renovatebot.com/merge-confidence/) | [![passing](https://badges.renovateapi.com/packages/maven/com.google.cloud:libraries-bom/26.3.0/compatibility-slim/26.2.0)](https://docs.renovatebot.com/merge-confidence/) | [![confidence](https://badges.renovateapi.com/packages/maven/com.google.cloud:libraries-bom/26.3.0/confidence-slim/26.2.0)](https://docs.renovatebot.com/merge-confidence/) |

---

### Configuration

📅 **Schedule**: Branch creation - At any time (no schedule defined), Automerge - At any time (no schedule defined).

🚦 **Automerge**: Disabled by config. Please merge this manually once you are satisfied.

♻ **Rebasing**: Renovate will not automatically rebase this PR, because other commits have been found.

🔕 **Ignore**: Close this PR and you won't be reminded about these updates again.

---

 - [ ] <!-- rebase-check -->If you want to rebase/retry this PR, check this box

---

This PR has been generated by [Mend Renovate](https://www.mend.io/free-developer-tools/renovate/). View repository job log [here](https://app.renovatebot.com/dashboard#github/googleapis/java-spanner).
<!--renovate-debug:eyJjcmVhdGVkSW5WZXIiOiIzNC4zMi4wIiwidXBkYXRlZEluVmVyIjoiMzQuMzIuMCJ9-->

gauravpurohit06 committed:
chore: Adding Kokoro Configurations to run integration tests against staging and devel environment. (#2236)

renovate-bot committed:
deps: update dependency org.graalvm.sdk:graal-sdk to v22.3.1 (#2238)

[![Mend Renovate](https://app.renovatebot.com/images/banner.svg)](https://renovatebot.com)

This PR contains the following updates:

| Package | Change | Age | Adoption | Passing | Confidence |
|---|---|---|---|---|---|
| [org.graalvm.sdk:graal-sdk](https://togithub.com/oracle/graal) | `22.3.0` -> `22.3.1` | [![age](https://badges.renovateapi.com/packages/maven/org.graalvm.sdk:graal-sdk/22.3.1/age-slim)](https://docs.renovatebot.com/merge-confidence/) | [![adoption](https://badges.renovateapi.com/packages/maven/org.graalvm.sdk:graal-sdk/22.3.1/adoption-slim)](https://docs.renovatebot.com/merge-confidence/) | [![passing](https://badges.renovateapi.com/packages/maven/org.graalvm.sdk:graal-sdk/22.3.1/compatibility-slim/22.3.0)](https://docs.renovatebot.com/merge-confidence/) | [![confidence](https://badges.renovateapi.com/packages/maven/org.graalvm.sdk:graal-sdk/22.3.1/confidence-slim/22.3.0)](https://docs.renovatebot.com/merge-confidence/) |

---

### Configuration

📅 **Schedule**: Branch creation - At any time (no schedule defined), Automerge - At any time (no schedule defined).

🚦 **Automerge**: Disabled by config. Please merge this manually once you are satisfied.

♻ **Rebasing**: Whenever PR becomes conflicted, or you tick the rebase/retry checkbox.

🔕 **Ignore**: Close this PR and you won't be reminded about this update again.

---

 - [ ] <!-- rebase-check -->If you want to rebase/retry this PR, check this box

---

This PR has been generated by [Mend Renovate](https://www.mend.io/free-developer-tools/renovate/). View repository job log [here](https://app.renovatebot.com/dashboard#github/googleapis/java-spanner).
<!--renovate-debug:eyJjcmVhdGVkSW5WZXIiOiIzNC45Ny42IiwidXBkYXRlZEluVmVyIjoiMzQuOTcuNiJ9-->

ko3a4ok committed:
feat: add support for Optimistic Concurrency Control (#2142)

* feat: add support for Optimistic Concurrency Control

Co-authored-by: Owl Bot <gcf-owl-bot[bot]@users.noreply.github.com>
Co-authored-by: Rajat Bhatta <93644539+rajatbhatta@users.noreply.github.com>

renovate-bot committed:
chore(deps): update dependency com.google.cloud:libraries-bom to v26.4.0 (#2240)

* chore(deps): update dependency com.google.cloud:libraries-bom to v26.4.0

* 🦉 Updates from OwlBot post-processor

See https://github.com/googleapis/repo-automation-bots/blob/main/packages/owl-bot/README.md

Co-authored-by: Owl Bot <gcf-owl-bot[bot]@users.noreply.github.com>

renovate-bot committed:
deps: update dependency com.google.cloud:google-cloud-trace to v2.8.0 (#2231)

[![Mend Renovate](https://app.renovatebot.com/images/banner.svg)](https://renovatebot.com)

This PR contains the following updates:

| Package | Change | Age | Adoption | Passing | Confidence |
|---|---|---|---|---|---|
| [com.google.cloud:google-cloud-trace](https://togithub.com/googleapis/google-cloud-java) | `2.7.0` -> `2.8.0` | [![age](https://badges.renovateapi.com/packages/maven/com.google.cloud:google-cloud-trace/2.8.0/age-slim)](https://docs.renovatebot.com/merge-confidence/) | [![adoption](https://badges.renovateapi.com/packages/maven/com.google.cloud:google-cloud-trace/2.8.0/adoption-slim)](https://docs.renovatebot.com/merge-confidence/) | [![passing](https://badges.renovateapi.com/packages/maven/com.google.cloud:google-cloud-trace/2.8.0/compatibility-slim/2.7.0)](https://docs.renovatebot.com/merge-confidence/) | [![confidence](https://badges.renovateapi.com/packages/maven/com.google.cloud:google-cloud-trace/2.8.0/confidence-slim/2.7.0)](https://docs.renovatebot.com/merge-confidence/) |

---

### Configuration

📅 **Schedule**: Branch creation - At any time (no schedule defined), Automerge - At any time (no schedule defined).

🚦 **Automerge**: Disabled by config. Please merge this manually once you are satisfied.

♻ **Rebasing**: Whenever PR becomes conflicted, or you tick the rebase/retry checkbox.

🔕 **Ignore**: Close this PR and you won't be reminded about this update again.

---

 - [ ] <!-- rebase-check -->If you want to rebase/retry this PR, check this box

---

This PR has been generated by [Mend Renovate](https://www.mend.io/free-developer-tools/renovate/). View repository job log [here](https://app.renovatebot.com/dashboard#github/googleapis/java-spanner).
<!--renovate-debug:eyJjcmVhdGVkSW5WZXIiOiIzNC43My4zIiwidXBkYXRlZEluVmVyIjoiMzQuNzMuMyJ9-->

release-please[bot] committed:
chore(main): release 6.35.1 (#2235)

:robot: I have created a release *beep* *boop*
---


## [6.35.1](https://togithub.com/googleapis/java-spanner/compare/v6.35.0...v6.35.1) (2023-01-18)


### Dependencies

* Update dependency com.google.cloud:google-cloud-monitoring to v3.9.0 ([#2230](https://togithub.com/googleapis/java-spanner/issues/2230)) ([717f70f](https://togithub.com/googleapis/java-spanner/commit/717f70f76f915e15a7283b32a83a6f4ac64fc931))
* Update dependency com.google.cloud:google-cloud-trace to v2.8.0 ([#2231](https://togithub.com/googleapis/java-spanner/issues/2231)) ([557ea16](https://togithub.com/googleapis/java-spanner/commit/557ea164ebf948cd78f937c6996fd21e9618d3ae))
* Update dependency org.graalvm.sdk:graal-sdk to v22.3.1 ([#2238](https://togithub.com/googleapis/java-spanner/issues/2238)) ([d5f5237](https://togithub.com/googleapis/java-spanner/commit/d5f52375394ef617f4fcb823937a374930f941e7))
* Update dependency org.junit.vintage:junit-vintage-engine to v5.9.2 ([#2223](https://togithub.com/googleapis/java-spanner/issues/2223)) ([3278f91](https://togithub.com/googleapis/java-spanner/commit/3278f9167b1b2688ed090a7dfd5874e88b8945a5))

---
This PR was generated with [Release Please](https://togithub.com/googleapis/release-please). See [documentation](https://togithub.com/googleapis/release-please#release-please).

release-please[bot] committed:
chore(main): release 6.35.2-SNAPSHOT (#2241)

:robot: I have created a release *beep* *boop*
---


### Updating meta-information for bleeding-edge SNAPSHOT release.

---
This PR was generated with [Release Please](https://togithub.com/googleapis/release-please). See [documentation](https://togithub.com/googleapis/release-please#release-please).

renovate-bot committed:
chore(deps): update dependency com.google.cloud:google-cloud-spanner to v6.35.1 (#2242)

* chore(deps): update dependency com.google.cloud:google-cloud-spanner to v6.35.1

* 🦉 Updates from OwlBot post-processor

See https://github.com/googleapis/repo-automation-bots/blob/main/packages/owl-bot/README.md

Co-authored-by: Owl Bot <gcf-owl-bot[bot]@users.noreply.github.com>

harshachinta committed:
feat: add samples for multi region CMEK

harshachinta committed:
feat: add unit tests for files under encryption

harshachinta committed:
Merge branch 'googleapis:main' into multiregion_cmek

harshachinta committed:
feat: add unit tests for multi region CMEK and code formatting

harshachinta committed:
Merge remote-tracking branch 'origin/multiregion_cmek' into multiregion_cmek

harshachinta commented:
Close local PR and raise to remote
